Lib/ufo2ft/filters/dottedCircleFilter.py

```diff
@@ -109,28 +109,28 @@ def __call__(self, font, glyphSet=None):
 
         self.set_context(font, glyphSet)
         added_glyph = False
-        dotted_circle = self.check_dotted_circle()
-        if not dotted_circle:
-            dotted_circle = self.draw_dotted_circle(glyphSet)
+        dotted_circle_glyph = self.check_dotted_circle()
+        if not dotted_circle_glyph:
+            dotted_circle_glyph = self.draw_dotted_circle(glyphSet)
             added_glyph = True
 
-        added_anchors = self.check_and_add_anchors(dotted_circle)
+        added_anchors = self.check_and_add_anchors(dotted_circle_glyph)
 
         if added_anchors:
-            self.ensure_base(dotted_circle)
+            self.ensure_base(dotted_circle_glyph)
 
         if added_glyph or added_anchors:
-            return [dotted_circle]
+            return [dotted_circle_glyph.name]
         else:
             return []
 
     def check_dotted_circle(self):
-        """Check for the presence of a dotted circle glyph and return its name"""
+        """Check for the presence of a dotted circle glyph and return it"""
         font = self.context.font
         dotted_circle = next((g.name for g in font if 0x25CC in g.unicodes), None)
         if dotted_circle:
             logger.debug("Found dotted circle glyph %s", dotted_circle)
-            return dotted_circle
+            return font[dotted_circle]
```

Comment:
Uh, by the way, do we want to fetch the glyph from the font or from the potentially different glyphSet? I'm confused now. @anthrotype

Reply:
Probably should be glyphSet, shouldn't it, in case we're dealing with a non-default layer?

Reply:
i'm not sure.. perhaps we don't want to add the dotted circle to "sparse" layers (i.e. DS sources pointing to same .ufo with layerName attribute) at all? it should be enough for it to be only on the full masters (arguably even enough on the default master).
I guess one can control that by adding the relevant lib key only to the ufos that need adding.
Actually, sparse layers don't even have their own distinct lib, we only check the font's lib for filters (not the layerinfo.plist).. 
I'm concerned that by adding an incomplete dotcircle glyph to a sparse layer we may get into GPOS varLib.merge errors with different set of anchors

Reply:
in any case, this should return the glyph from the glyphSet, because even when it's not a sparse layer, the glyphSet is a deepcopy of the original glyphs, which is what the various filters can modify in-place, not the original glyphs themselves.